src/core/buffers/buffer_orchestrator.ts

```diff
@@ -257,24 +258,29 @@ export default function BufferOrchestrator(
       filter(({ currentTime, wantedTimeOffset }) => {
         return enableOutOfBoundsCheck &&
                manifest.getPeriodForTime(wantedTimeOffset +
-                                           currentTime) !== undefined &&
+                                         currentTime) !== undefined &&
                isOutOfPeriodList(wantedTimeOffset + currentTime);
       }),
-      tap(({ currentTime, wantedTimeOffset }) => {
+      mergeMap(({ currentTime, isPaused, wantedTimeOffset }) => {
         log.info("BO: Current position out of the bounds of the active periods," +
                  "re-creating buffers.",
                  bufferType,
                  currentTime + wantedTimeOffset);
         enableOutOfBoundsCheck = false;
-        destroyBuffers$.next();
-      }),
-      mergeMap(({ currentTime, wantedTimeOffset }) => {
+        const outOfBoundsPeriod = manifest.getPeriodForTime(wantedTimeOffset +
+                                                            currentTime);
+        if (shouldReloadAtEachPeriodChange && outOfBoundsPeriod !== basePeriod) {
+          return observableOf({ type: "needs-media-source-reload" as const,
+                                value: { currentTime: currentTime + 0.01,
+                                         isPaused } });
+        }
         const newInitialPeriod = manifest
           .getPeriodForTime(currentTime + wantedTimeOffset);
         if (newInitialPeriod == null) {
           throw new MediaError("MEDIA_TIME_NOT_FOUND",
                                "The wanted position is not found in the Manifest.");
         }
+        destroyBuffers$.next();
```

Comment:
There is a risk there because `destroyBuffers$.next()` actually performs unknown side-effects.

Those side-effects could lead to annoying cases, like provoke an event sent by the API and the user deciding to stop on such event. Here we would be calling `launchConsecutiveBuffersForPeriod` while the stream is already stopped, possibly leading to hard to understand bugs.

That's always a problem when using subjects which is why those operations should generally always be done last in an operator, ensuring that any side-effects stopping that observable are catched before continuing the logic from that same observable.

Reply:
Ok i understand.